common/src/uint/impls.rs

```diff
@@ -691,11 +691,11 @@ mod tests {
     pub fn uint256_display_test() {
         assert_eq!(
             format!("{:?}", Uint256::from_u64(0xDEADBEEF)),
-            "0x00000000000000000000000000000000000000000000000000000000deadbeef"
+            "0xefbeadde00000000000000000000000000000000000000000000000000000000"
         );
```

Comment:
I think `Uint256` values should be printed in BE, so the printed string stays in the natural numeric notation. Same as `u64` and similar.

Reply:
The problem here is that rust treats both to_string and display similarly and Display implements ToString automatically... which can easily lead to confusion.

Imagine a dev working, not worrying about endianness. They want a function to convert to hex. They try to_string, and it writes in hex. They use it, shrug it off and contiue. Big mistake if not caught in a PR.

How do we prevent big mistakes? We make them impossible. 

The way I think is: There should be exclusively one way to print things, so that people don't mix multiple ways up.

Reply:
I also think it would be better to keep printing our `UintXXX` types in the natural order (like it was before) for consistency with built-in types.

Reply:
> Imagine a dev working, not worrying about endianness. They want a function to convert to hex. They try to_string, and it writes in hex. They use it, shrug it off and contiue. Big mistake if not caught in a PR.

I think we can get both behave consistently as long as the conversion methods between `Uint256` and the likes of `H256`, `[u8; 32]` and such consistently perform a BE conversion.